src/main.rs

```diff
@@ -348,57 +346,70 @@ fn get_config_and_cache_paths() -> (PathBuf, PathBuf) {
 }
 
 #[cfg(not(feature = "networking"))]
-/// Always returns false without the `networking` feature.
-fn check_for_updates() -> bool {
-	false
+mod update {
+	/// Always returns false without the `networking` feature.
+	pub fn check_for_updates() -> bool {
+		false
+	}
 }
 
 #[cfg(feature = "networking")]
-/// Returns true if updates are available.
-fn check_for_updates() -> bool {
-	use crate::version::Version;
-	use std::str::FromStr;
-
+mod version;
+#[cfg(feature = "networking")]
+mod update {
 	use serde::Deserialize;
 
 	#[derive(Deserialize)]
 	struct ReleaseInfoJson {
 		tag_name: String,
 	}
 
-	let client = match reqwest::blocking::Client::builder().user_agent("emulsion").build() {
-		Ok(c) => c,
-		Err(e) => {
-			println!("Could not build client for version request: {}", e);
-			return false;
+	mod errors {
+		error_chain! {
+			foreign_links {
+				Io(std::io::Error);
+				JsonError(serde_json::error::Error);
+			}
 		}
```

Comment:
You define this custom error type here, and then you use it as the return value of the `latest_release` function that you call like this:
```rust
if let Ok(info) = latest_release() {
   //...
}
```
And that's it, you don't do anything with the error. I'm not sure if this is intentional. You could print the error to the stderr or alternatively just return an `Option<ReleaseInfoJson>` from the function so you don't create a type that you end up not using.

Reply:
Good catch, signaling of JSON errors in `check_for_updates` was lost in the process, and we're silently returning `false` right now. I will adjust `check_for_updates` accordingly.

Reply:
Good catch, signaling of error details in `check_for_updates` was lost in the process, and we're silently returning `false` right now. I will adjust `check_for_updates` accordingly.

Reply:
I don't think that returning an error from `check_for_updates` is benefitial. We would print that error message to the stderr anyways and there's nothing else we would want to do with that error. So it's best to keep the boilerplate to a minimum and just print the error within `check_for_updates`, and return false.

Reply:
Sorry for the confusion, That's not what I meant and not what I just pushed to the branch. I meant that reporting of parsing errors was lost in the migration. It's now been restored while still retaining the original signature of `check_for_updates`. The two helper functions may return errors, different types of errors, and we catch and report either error condition in `check_for_updates`.

Reply:
Sorry for the confusion, that's not what I meant and not what I just pushed to the branch. I meant that reporting of parsing errors was lost in the migration, which has now been restored while still retaining the original signature of `check_for_updates`. The two helper functions may return errors, different types of errors, and we catch and report either error condition in `check_for_updates`.